src/pages/Menu.tsx

```diff
@@ -64,76 +63,67 @@ const Menu = () => {
     <div>
       {/* Conditional Search Area and Button Group */}
       <div className="button-group">
-        {view === "categories" && (
-          <div className="search-area">
-            <input
-              type="text"
-              placeholder="Search for Categories"
-              value={searchQuery}
-              onChange={handleSearch}
-            />
-          </div>
+        {view === View.Categories && (
+          <SearchArea
+            placeholder="Search for Categories"
+            onChange={handleSearch}
+            searchQuery={searchQuery}
+          />
         )}
         <button
-          onClick={() => setView("categories")}
-          className={`menu-button ${view === "categories" ? "active" : ""}`}
+          onClick={() => setView(View.Categories)}
+          className={`menu-button ${view === View.Categories ? "active" : ""}`}
         >
           Categories
         </button>
 
-        {view === "areas" && (
-          <div className="search-area">
-            <input
-              type="text"
-              placeholder="Search for National Dishes"
-              value={searchQuery}
-              onChange={handleSearch}
-            />
-          </div>
+        {view === View.Areas && (
+          <SearchArea
+            placeholder="Search for National Dishes"
+            onChange={handleSearch}
+            searchQuery={searchQuery}
+          />
         )}
         <button
-          onClick={() => setView("areas")}
-          className={`menu-button ${view === "areas" ? "active" : ""}`}
+          onClick={() => setView(View.Areas)}
+          className={`menu-button ${view === View.Areas ? "active" : ""}`}
         >
           National Dishes
         </button>
 
-        {view === "allMeals" && (
-          <div className="search-area">
-            <input
-              type="text"
-              placeholder="Search All Meals"
-              value={searchQuery}
-              onChange={handleSearch}
-            />
-          </div>
+        {view === View.AllMeals && (
+          <SearchArea
+            placeholder="Search All Meals"
+            onChange={handleSearch}
+            searchQuery={searchQuery}
+          />
         )}
         <button
-          onClick={() => setView("allMeals")}
-          className={`menu-button ${view === "allMeals" ? "active" : ""}`}
+          onClick={() => setView(View.AllMeals)}
+          className={`menu-button ${view === View.AllMeals ? "active" : ""}`}
         >
           All Meals
         </button>
       </div>
 
       <div className="menu-grid">
-        {view === "categories" &&
+        {view === View.Categories &&
```

Comment:
put it in ? tipa view===View.Categories? <> some code block<>: view===View.Areas?<> some block code about areas <>: <> last option <>